activestorage/test/test_helper.rb

```diff
@@ -54,6 +54,12 @@ def create_blob_before_direct_upload(filename: "hello.txt", byte_size:, checksum
       ActiveStorage::Blob.create_before_direct_upload! filename: filename, byte_size: byte_size, checksum: checksum, content_type: content_type
     end
 
+    def simulate_direct_upload_blob
+      create_blob_before_direct_upload(filename: "racecar.jpg", content_type: "application/octet-stream", byte_size: 1124062, checksum: "7GjDDNEQb4mzMzsW+MS0JQ==").tap do |blob|
+        ActiveStorage::Blob.service.upload(blob.key, file_fixture("racecar.jpg").open)
+      end
+    end
```

Comment:
Parameterize the blob attributes so this method may be used for other file fixtures without modification:

```ruby
def directly_upload_file_blob(filename: "racecar.jpg", content_type: "image/jpeg")
  file = file_fixture(filename)
  byte_size = file.size
  checksum = Digest::MD5.file(file).base64digest

  create_blob_before_direct_upload(filename: filename, byte_size: byte_size, checksum: checksum, content_type: content_type).tap do |blob|
    ActiveStorage::Blob.service.upload(blob.key, file.open)
  end
end
```